# **Setting up Conda on Google colab**

In [ ]:
!wget https://repo.anaconda.com/miniconda/Miniconda3-py37_4.8.2-Linux-x86_64.sh
!chmod +x Miniconda3-py37_4.8.2-Linux-x86_64.sh
!bash ./Miniconda3-py37_4.8.2-Linux-x86_64.sh -b -f -p /usr/local
import sys
sys.path.append('/usr/local/lib/python3.7/site-packages/')
!conda install -c rdkit rdkit -y
!conda install pytorch=1.7.1 torchvision=0.8.2 cudatoolkit=11.0 -c pytorch -y
!conda install cython scipy -y
!pip install pycocotools
!pip install opencv-python
!pip install wandb
!pip install --upgrade --no-cache-dir gdown --pre

--2022-05-03 16:18:39--  https://repo.anaconda.com/miniconda/Miniconda3-py37_4.8.2-Linux-x86_64.sh
Resolving repo.anaconda.com (repo.anaconda.com)... 104.16.131.3, 104.16.130.3, 2606:4700::6810:8203, ...
Connecting to repo.anaconda.com (repo.anaconda.com)|104.16.131.3|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 85055499 (81M) [application/x-sh]
Saving to: ‘Miniconda3-py37_4.8.2-Linux-x86_64.sh’

Miniconda3-py37_4.8 100%[===================>]  81.12M  13.5MB/s    in 7.8s    

2022-05-03 16:18:47 (10.4 MB/s) - ‘Miniconda3-py37_4.8.2-Linux-x86_64.sh’ saved [85055499/85055499]

PREFIX=/usr/local
Unpacking payload ...
Solving environment: - \ done

## Package Plan ##

  environment location: /usr/local

  added / updated specs:
    - _libgcc_mutex==0.1=main
    - asn1crypto==1.3.0=py37_0
    - ca-certificates==2020.1.1=0
    - certifi==2019.11.28=py37_0
    - cffi==1.14.0=py37h2e261b9_0
    - chardet==3.0.4=py37_1003
    - conda-package-handling==1.6.0=py37

     |████████████████████████████████| 60.5 MB 1.3 MB/s 
     |████████████████████████████████| 1.8 MB 14.6 MB/s 
     |████████████████████████████████| 280 kB 66.0 MB/s 
     |████████████████████████████████| 181 kB 63.4 MB/s 
     |████████████████████████████████| 1.0 MB 51.3 MB/s 
     |████████████████████████████████| 96 kB 7.6 MB/s 
     |████████████████████████████████| 596 kB 57.7 MB/s 
     |████████████████████████████████| 144 kB 60.0 MB/s 
     |████████████████████████████████| 63 kB 2.2 MB/s 
  Created wheel for pathtools: filename=pathtools-0.1.2-py3-none-any.whl size=8784 sha256=cc2843585ee1000c10f37a81c02ba3af524440275ab9683235a12ddc78469d34
  Stored in directory: /root/.cache/pip/wheels/3e/31/09/fa59cef12cdcfecc627b3d24273699f390e71828921b2cbba2
  Created wheel for promise: filename=promise-2.3-py3-none-any.whl size=21495 sha256=2b4d7eea298ecdc61201e13b43178b0cc77cdac68ee3c428bc4d81d2d80a5e35
  Stored in directory: /root/.cache/pip/wheels/29/93/c6/762e359f8cb6a5

# **Connecting to Google drive**

In [ ]:
# Connecting to gdrive
from google.colab import drive
drive.mount('/content/drive')

# Access HOI folder
%cd /content/drive/MyDrive/HOI

# **Cloning HOTR repo**

In [ ]:
# Cloning HOTR repo from GitHub
!git clone https://github.com/kakaobrain/HOTR.git
# Change to HOTR dir
%cd HOTR

Cloning into 'HOTR'...
remote: Enumerating objects: 152, done.
remote: Counting objects: 100% (152/152), done.
remote: Compressing objects: 100% (101/101), done.
remote: Total 152 (delta 67), reused 130 (delta 48), pack-reused 0
Receiving objects: 100% (152/152), 716.39 KiB | 4.48 MiB/s, done.
Resolving deltas: 100% (67/67), done.
/content/drive/MyDrive/HOI/HOTR


# **Setting up Datasets**

### From my Google **drive**

In [ ]:
# Downloading and setting up V-COCO
%cd /content/drive/MyDrive/HOI/HOTR
!git clone --recursive https://github.com/s-gupta/v-coco.git ./v-coco/
%cd ./v-coco/coco
%mkdir images

!unzip /content/drive/MyDrive/train2014.zip -d images/
!unzip /content/drive/MyDrive/val2014.zip -d images/
!unzip /content/drive/MyDrive/test2014.zip -d images/
!unzip /content/drive/MyDrive/annotations_trainval2014.zip

# Pick out annotations from the COCO annotations to allow faster loading in V-COCO
%cd ../
!python script_pick_annotations.py coco/annotations

# Build
%cd /content/drive/MyDrive/HOI/HOTR/v-coco/coco/PythonAPI
!make
%cd /content/drive/MyDrive/HOI/HOTR/v-coco
!make

Streaming output truncated to the last 5000 lines.
 extracting: images/test2014/COCO_test2014_000000225640.jpg  
 extracting: images/test2014/COCO_test2014_000000202014.jpg  
 extracting: images/test2014/COCO_test2014_000000074676.jpg  
 extracting: images/test2014/COCO_test2014_000000011023.jpg  
 extracting: images/test2014/COCO_test2014_000000421945.jpg  
 extracting: images/test2014/COCO_test2014_000000273821.jpg  
 extracting: images/test2014/COCO_test2014_000000080301.jpg  
 extracting: images/test2014/COCO_test2014_000000282955.jpg  
 extracting: images/test2014/COCO_test2014_000000154760.jpg  
 extracting: images/test2014/COCO_test2014_000000305164.jpg  
 extracting: images/test2014/COCO_test2014_000000112652.jpg  
 extracting: images/test2014/COCO_test2014_000000167917.jpg  
 extracting: images/test2014/COCO_test2014_000000259779.jpg  
 extracting: images/test2014/COCO_test2014_000000403391.jpg  
 extracting: images/test2014/COCO_test2014_000000077358.jpg  
 extracting: images

In [ ]:
# Downloading and setting up HICO-DET
%cd /content/drive/MyDrive/HOI/HOTR
!tar -xvzf /content/drive/MyDrive/hico_20160224_det.tar.gz

# downloading annotations for hico-det
%cd /content/drive/MyDrive/HOI/HOTR/hico_20160224_det
%mkdir annotations
%cd /content/drive/MyDrive/HOI/HOTR/hico_20160224_det/annotations

!gdown --id 1yoWvEC-G893GzPZ8mx-wlWdvK7w-2l-_
!gdown --id 1GHUMSlU-fpOpJk3lvegORUS_hxvRvfXB
!gdown --id 1jPsJPNXJNWeVF5zvCAvvGata4afl1bPA

# downloading hico_list_vb
%cd /content/drive/MyDrive/HOI/HOTR/hico_20160224_det
!gdown --id 1EeHNHuYyJI-qqDk_-5nay7Mb07tzZLsl

# renaming it as needed in the script
%mv hico_list_vb.txt list_action.txt

Streaming output truncated to the last 5000 lines.
hico_20160224_det/images/test2015/HICO_test2015_00000968.jpg
hico_20160224_det/images/test2015/HICO_test2015_00002723.jpg
hico_20160224_det/images/test2015/HICO_test2015_00000149.jpg
hico_20160224_det/images/test2015/HICO_test2015_00005540.jpg
hico_20160224_det/images/test2015/HICO_test2015_00002445.jpg
hico_20160224_det/images/test2015/HICO_test2015_00007462.jpg
hico_20160224_det/images/test2015/HICO_test2015_00003166.jpg
hico_20160224_det/images/test2015/HICO_test2015_00008375.jpg
hico_20160224_det/images/test2015/HICO_test2015_00004788.jpg
hico_20160224_det/images/test2015/HICO_test2015_00006776.jpg
hico_20160224_det/images/test2015/HICO_test2015_00002754.jpg
hico_20160224_det/images/test2015/HICO_test2015_00007288.jpg
hico_20160224_det/images/test2015/HICO_test2015_00009110.jpg
hico_20160224_det/images/test2015/HICO_test2015_00003776.jpg
hico_20160224_det/images/test2015/HICO_test2015_00001395.jpg
hico_20160224_det/images/test2015/

### From original servers

In [ ]:
# Download V-COCO
%cd /content/drive/MyDrive/HOI/HOTR
!git clone --recursive https://github.com/s-gupta/v-coco.git ./v-coco/
%cd ./v-coco/coco

!wget -N http://images.cocodataset.org/zips/train2014.zip
!wget -N http://images.cocodataset.org/zips/val2014.zip
!wget -N http://images.cocodataset.org/zips/test2014.zip
!wget -N http://images.cocodataset.org/annotations/annotations_trainval2014.zip

%mkdir images

!unzip train2014.zip -d images/
!unzip val2014.zip -d images/
!unzip test2014.zip -d images/
!unzip annotations_trainval2014.zip

%rm train2014.zip
%rm val2014.zip
%rm test2014.zip
%rm annotations_trainval2014.zip

# Pick out annotations from the COCO annotations to allow faster loading in V-COCO
%cd ../
!python script_pick_annotations.py coco/annotations

# Build
%cd /content/drive/MyDrive/HOI/HOTR/v-coco/coco/PythonAPI
!make
%cd /content/drive/MyDrive/HOI/HOTR/v-coco
!make

In [ ]:
# Downloading HICO-DET
%cd /content/drive/MyDrive/HOI/HOTR
!gdown --id 1QZcJmGVlF9f4h-XLWe9Gkmnmj2z1gSnk

# uncompress hico-det & then remove compressed file
!tar -xvzf hico_20160224_det.tar.gz
%rm hico_20160224_det.tar.gz

# downloading annotations for hico-det
%cd /content/drive/MyDrive/HOI/HOTR/hico_20160224_det
%mkdir annotations
%cd /content/drive/MyDrive/HOI/HOTR/hico_20160224_det/annotations

!gdown --id 1yoWvEC-G893GzPZ8mx-wlWdvK7w-2l-_
!gdown --id 1GHUMSlU-fpOpJk3lvegORUS_hxvRvfXB
!gdown --id 1jPsJPNXJNWeVF5zvCAvvGata4afl1bPA

# downloading hico_list_vb
%cd /content/drive/MyDrive/HOI/HOTR/hico_20160224_det
!gdown --id 1EeHNHuYyJI-qqDk_-5nay7Mb07tzZLsl

# renaming it as needed in the script
%mv hico_list_vb.txt list_action.txt

# **Training the model**

### V-COCO single-gpu train

In [ ]:
%cd /content/drive/MyDrive/HOI/HOTR
!python main.py \
  --group_name KakaoBrain_HOTR_vcoco \
  --run_name vcoco_single_run_000001 \
  --HOIDet \
  --validate \
  --share_enc \
  --pretrained_dec \
  --lr 1e-4 \
  --num_hoi_queries 16 \
  --set_cost_idx 10 \
  --hoi_act_loss_coef 10 \
  --hoi_eos_coef 0.1 \
  --temperature 0.05 \
  --epochs 30 \
  --no_aux_loss \
  --hoi_aux_loss \
  --dataset_file vcoco \
  --frozen_weights https://dl.fbaipublicfiles.com/detr/detr-r50-e632da11.pth \
  --data_path v-coco  \
  --output_dir checkpoints/vcoco/

/content/drive/MyDrive/HOI/HOTR
Not using distributed mode
Freeze weights for detector
loading annotations into memory...
Done (t=1.79s)
creating index...
index created!
loading annotations into memory...
Done (t=2.10s)
creating index...
index created!

[Logger] DETR Arguments:
	lr: 0.0001
	lr_backbone: 1e-05
	lr_drop: 80
	frozen_weights: https://dl.fbaipublicfiles.com/detr/detr-r50-e632da11.pth
	backbone: resnet50
	dilation: False
	position_embedding: sine
	enc_layers: 6
	dec_layers: 6
	num_queries: 100
	dataset_file: vcoco

[Logger] DETR_HOI Arguments:
	hoi_dec_layers: 6
	hoi_nheads: 8
	hoi_dim_feedforward: 2048
	hoi_idx_loss_coef: 1
	hoi_act_loss_coef: 10.0
	hoi_eos_coef: 0.1
	object_threshold: 0
Downloading: "https://download.pytorch.org/models/resnet50-19c8e357.pth" to /root/.cache/torch/hub/checkpoints/resnet50-19c8e357.pth
100% 97.8M/97.8M [00:00<00:00, 115MB/s] 

[Logger] Number of params:  9879582
Downloading: "https://dl.fbaipublicfiles.com/detr/detr-r50-e632da11.pth" to /roo

In [ ]:
# Training in batches to avoid Google colab time out issues
%cd /content/drive/MyDrive/HOI/HOTR
!python main.py \
  --group_name GR_HOTR_vcoco \
  --run_name vcoco_single_run_000001 \
  --HOIDet \
  --validate \
  --share_enc \
  --pretrained_dec \
  --lr 1e-4 \
  --num_hoi_queries 16 \
  --set_cost_idx 10 \
  --hoi_act_loss_coef 10 \
  --hoi_eos_coef 0.1 \
  --temperature 0.05 \
  --epochs 30 \
  --no_aux_loss \
  --hoi_aux_loss \
  --dataset_file vcoco \
  --frozen_weights https://dl.fbaipublicfiles.com/detr/detr-r50-e632da11.pth \
  --data_path v-coco  \
  --output_dir checkpoints/vcoco/ \
  --resume checkpoints/vcoco/GR_HOTR_vcoco/vcoco_single_run_000001/checkpoint.pth \
  --start_epoch 25

/content/drive/MyDrive/HOI/HOTR
Not using distributed mode
Freeze weights for detector
loading annotations into memory...
Done (t=1.58s)
creating index...
index created!
loading annotations into memory...
Done (t=2.09s)
creating index...
index created!

[Logger] DETR Arguments:
	lr: 0.0001
	lr_backbone: 1e-05
	lr_drop: 80
	frozen_weights: https://dl.fbaipublicfiles.com/detr/detr-r50-e632da11.pth
	backbone: resnet50
	dilation: False
	position_embedding: sine
	enc_layers: 6
	dec_layers: 6
	num_queries: 100
	dataset_file: vcoco

[Logger] DETR_HOI Arguments:
	hoi_dec_layers: 6
	hoi_nheads: 8
	hoi_dim_feedforward: 2048
	hoi_idx_loss_coef: 1
	hoi_act_loss_coef: 10.0
	hoi_eos_coef: 0.1
	object_threshold: 0

[Logger] Number of params:  9879582

>>> Epoch #27
Epoch [27/30]  [ 539/2692]  eta: 0:34:13  lr: 0.000100  loss: 31.9953 (32.0109)  loss_hidx: 0.1192 (0.2865)  loss_oidx: 0.4388 (0.7423)  loss_act: 4.3414 (3.9939)  loss_hidx_0: 0.0898 (0.3363)  loss_oidx_0: 0.7833 (0.8445)  loss_act_0: 4.7

### HICO-DET single-gpu train

In [ ]:
# Training 30 epochs (took around 45 hours) (trained in smaller batch)
!python main.py \
		--group_name GR_HOTR_hicodet \
		--run_name hicodet_single_run_000001 \
		--HOIDet \
		--validate \
		--share_enc \
		--pretrained_dec \
		--lr 1e-4 \
		--num_hoi_queries 16 \
		--set_cost_idx 20 \
		--hoi_act_loss_coef 10 \
		--hoi_eos_coef 0.1 \
		--temperature 0.2 \
    --epochs 30 \
		--no_aux_loss \
		--hoi_aux_loss \
		--dataset_file hico-det \
		--frozen_weights https://dl.fbaipublicfiles.com/detr/detr-r50-e632da11.pth \
		--data_path hico_20160224_det \
		--output_dir checkpoints/hico_det/ \
    --resume checkpoints/hico_det/GR_HOTR_hicodet/hicodet_single_run_000001/checkpoint.pth \
    --start_epoch 28 \
    --num_workers 2

Not using distributed mode
Freeze weights for detector

[Logger] DETR Arguments:
	lr: 0.0001
	lr_backbone: 1e-05
	lr_drop: 80
	frozen_weights: https://dl.fbaipublicfiles.com/detr/detr-r50-e632da11.pth
	backbone: resnet50
	dilation: False
	position_embedding: sine
	enc_layers: 6
	dec_layers: 6
	num_queries: 100
	dataset_file: hico-det

[Logger] DETR_HOI Arguments:
	hoi_dec_layers: 6
	hoi_nheads: 8
	hoi_dim_feedforward: 2048
	hoi_idx_loss_coef: 1
	hoi_act_loss_coef: 10.0
	hoi_eos_coef: 0.1
	object_threshold: 0

[Logger] Number of params:  9902198

>>> Epoch #29
Epoch [29/30]  [ 3764/18816]  eta: 0:52:10  lr: 0.000100  loss: 64.9584 (69.2504)  loss_hidx: 2.5242 (2.4521)  loss_oidx: 3.0889 (2.8426)  loss_act: 3.6717 (4.2724)  loss_tgt: 0.6017 (1.4376)  loss_hidx_0: 2.5770 (2.5301)  loss_oidx_0: 3.1411 (2.9306)  loss_act_0: 5.6110 (5.5211)  loss_tgt_0: 0.6104 (1.5314)  loss_hidx_1: 2.3553 (2.4918)  loss_oidx_1: 3.0424 (2.8939)  loss_act_1: 4.6833 (5.0433)  loss_tgt_1: 0.3419 (1.5076)  loss_

# **Testing the model**

### Preparing for testing

In [ ]:
# making directory for checkpoints
%cd /content/drive/MyDrive/HOI/HOTR
%mkdir checkpoints
%cd checkpoints
%mkdir vcoco
%mkdir hico_det

%cd /content/drive/MyDrive/HOI/HOTR/checkpoints/vcoco
!wget -N https://arena.kakaocdn.net/brainrepo/hotr/vcoco_q16.pth

%cd /content/drive/MyDrive/HOI/HOTR/checkpoints/hico_det
!!wget -N https://arena.kakaocdn.net/brainrepo/hotr/hico_q16.pth

### Testing using author's weights

In [ ]:
# Testing the model on v-coco
%cd /content/drive/MyDrive/HOI/HOTR
!make vcoco_single_test

/content/drive/MyDrive/HOI/HOTR
python main.py \
	--HOIDet \
	--share_enc \
	--pretrained_dec \
	--num_hoi_queries 16 \
	--object_threshold 0 \
	--temperature 0.05 \
	--no_aux_loss \
	--eval \
	--dataset_file vcoco \
	--data_path v-coco \
	--resume checkpoints/vcoco/vcoco_q16.pth
Not using distributed mode
loading annotations into memory...
Done (t=1.56s)
creating index...
index created!
loading annotations into memory...
Done (t=1.88s)
creating index...
index created!

[Logger] DETR Arguments:
	lr: 0.0001
	lr_backbone: 1e-05
	lr_drop: 80
	frozen_weights: None
	backbone: resnet50
	dilation: False
	position_embedding: sine
	enc_layers: 6
	dec_layers: 6
	num_queries: 100
	dataset_file: vcoco

[Logger] DETR_HOI Arguments:
	hoi_dec_layers: 6
	hoi_nheads: 8
	hoi_dim_feedforward: 2048
	hoi_idx_loss_coef: 1
	hoi_act_loss_coef: 1
	hoi_eos_coef: 0.1
	object_threshold: 0.0

[Logger] Number of params:  51181950
Evaluation Inference (V-COCO)  [2462/2462]  eta: 0:00:00    time: 0.2647  data: 0.1425

In [ ]:
# Testing the model on hico-det
%cd /content/drive/MyDrive/HOI/HOTR
!make hico_single_test

/content/drive/MyDrive/HOI/HOTR
python main.py \
	--HOIDet \
	--share_enc \
	--pretrained_dec \
	--num_hoi_queries 16 \
	--object_threshold 0 \
	--temperature 0.2 \
	--no_aux_loss \
	--eval \
	--dataset_file hico-det \
	--data_path hico_20160224_det \
	--resume checkpoints/hico_det/hico_q16.pth
Not using distributed mode

[Logger] DETR Arguments:
	lr: 0.0001
	lr_backbone: 1e-05
	lr_drop: 80
	frozen_weights: None
	backbone: resnet50
	dilation: False
	position_embedding: sine
	enc_layers: 6
	dec_layers: 6
	num_queries: 100
	dataset_file: hico-det

[Logger] DETR_HOI Arguments:
	hoi_dec_layers: 6
	hoi_nheads: 8
	hoi_dim_feedforward: 2048
	hoi_idx_loss_coef: 1
	hoi_act_loss_coef: 1
	hoi_eos_coef: 0.1
	object_threshold: 0.0

[Logger] Number of params:  51204566
Evaluation Inference (HICO-DET)  [4773/4773]  eta: 0:00:00    time: 0.2745  data: 0.1820  max mem: 1193
[stats] Total Time (test) : 0:21:28 (0.2700 s / it)
[stats] HOI Recognition Time (avg) : 0.4528 ms
[stats] Score Matrix Generation

### Testing on my trained model

In [ ]:
# V-COCO
%cd /content/drive/MyDrive/HOI/HOTR
!python main.py \
		--HOIDet \
		--share_enc \
		--pretrained_dec \
		--num_hoi_queries 16 \
		--object_threshold 0 \
		--temperature 0.05 \
		--no_aux_loss \
		--eval \
		--dataset_file vcoco \
		--data_path v-coco \
		--resume checkpoints/vcoco/GR_HOTR_vcoco/vcoco_single_run_000001/best.pth

/content/drive/MyDrive/HOI/HOTR
Not using distributed mode
loading annotations into memory...
Done (t=1.77s)
creating index...
index created!
loading annotations into memory...
Done (t=1.97s)
creating index...
index created!

[Logger] DETR Arguments:
	lr: 0.0001
	lr_backbone: 1e-05
	lr_drop: 80
	frozen_weights: None
	backbone: resnet50
	dilation: False
	position_embedding: sine
	enc_layers: 6
	dec_layers: 6
	num_queries: 100
	dataset_file: vcoco

[Logger] DETR_HOI Arguments:
	hoi_dec_layers: 6
	hoi_nheads: 8
	hoi_dim_feedforward: 2048
	hoi_idx_loss_coef: 1
	hoi_act_loss_coef: 1
	hoi_eos_coef: 0.1
	object_threshold: 0.0

[Logger] Number of params:  51181950
Evaluation Inference (V-COCO)  [2462/2462]  eta: 0:00:00    time: 0.1653  data: 0.0130  max mem: 2976
[stats] Total Time (test) : 0:07:21 (0.1794 s / it)
[stats] HOI Recognition Time (avg) : 4.5986 ms
[stats] Distributed Gathering Time : 0:00:00
[stats] Score Matrix Generation completed!!          

============= AP (Role scenario_1)

In [ ]:
# V-COCO (older test)
%cd /content/drive/MyDrive/HOI/HOTR
!python main.py \
		--HOIDet \
		--share_enc \
		--pretrained_dec \
		--num_hoi_queries 16 \
		--object_threshold 0 \
		--temperature 0.05 \
		--no_aux_loss \
		--eval \
		--dataset_file vcoco \
		--data_path v-coco \
		--resume checkpoints/vcoco/KakaoBrain_HOTR_vcoco/vcoco_single_run_000001/best.pth

/content/drive/MyDrive/HOI/HOTR
Not using distributed mode
loading annotations into memory...
Done (t=1.56s)
creating index...
index created!
loading annotations into memory...
Done (t=1.90s)
creating index...
index created!

[Logger] DETR Arguments:
	lr: 0.0001
	lr_backbone: 1e-05
	lr_drop: 80
	frozen_weights: None
	backbone: resnet50
	dilation: False
	position_embedding: sine
	enc_layers: 6
	dec_layers: 6
	num_queries: 100
	dataset_file: vcoco

[Logger] DETR_HOI Arguments:
	hoi_dec_layers: 6
	hoi_nheads: 8
	hoi_dim_feedforward: 2048
	hoi_idx_loss_coef: 1
	hoi_act_loss_coef: 1
	hoi_eos_coef: 0.1
	object_threshold: 0.0

[Logger] Number of params:  51181950
Evaluation Inference (V-COCO)  [2462/2462]  eta: 0:00:00    time: 0.1290  data: 0.0115  max mem: 2976
[stats] Total Time (test) : 0:05:38 (0.1374 s / it)
[stats] HOI Recognition Time (avg) : 0.6921 ms
[stats] Distributed Gathering Time : 0:00:00
[stats] Score Matrix Generation completed!!          

============= AP (Role scenario_1)

In [ ]:
# HICO-DET
%cd /content/drive/MyDrive/HOI/HOTR
!python main.py \
		--HOIDet \
		--share_enc \
		--pretrained_dec \
		--num_hoi_queries 16 \
		--object_threshold 0 \
		--temperature 0.2 \
		--no_aux_loss \
		--eval \
		--dataset_file hico-det \
		--data_path hico_20160224_det \
		--resume checkpoints/hico_det/KakaoBrain_HOTR_hicodet/hicodet_single_run_000001/best.pth

/content/drive/MyDrive/HOI/HOTR
Not using distributed mode

[Logger] DETR Arguments:
	lr: 0.0001
	lr_backbone: 1e-05
	lr_drop: 80
	frozen_weights: None
	backbone: resnet50
	dilation: False
	position_embedding: sine
	enc_layers: 6
	dec_layers: 6
	num_queries: 100
	dataset_file: hico-det

[Logger] DETR_HOI Arguments:
	hoi_dec_layers: 6
	hoi_nheads: 8
	hoi_dim_feedforward: 2048
	hoi_idx_loss_coef: 1
	hoi_act_loss_coef: 1
	hoi_eos_coef: 0.1
	object_threshold: 0.0
Downloading: "https://download.pytorch.org/models/resnet50-19c8e357.pth" to /root/.cache/torch/hub/checkpoints/resnet50-19c8e357.pth
100% 97.8M/97.8M [00:00<00:00, 233MB/s]

[Logger] Number of params:  51204566

[stats] Total Time (test) : 0:45:40 (0.5743 s / it)
[stats] HOI Recognition Time (avg) : 1.5592 ms
[stats] Score Matrix Generation completed!!          
| mAP (full)		: 22.24
| mAP (rare)		: 20.81
| mAP (non-rare)	: 22.67


# Results

Author's model (100 epochs): 
  
1.   V-COCO: 58.99, 63.90
2.   HICO-DET: Full(23.74), Rare(22.52), Non-rare(24.10)

My trained model (30 epochs):

1.   V-COCO: 55.83, 60.72
2.   HICO-DET: Full(22.24), Rare(20.81), Non-rare(22.67)

### Old training block:

In [ ]:
# Timed out during traning
!python main.py \
		--group_name KakaoBrain_HOTR_hicodet \
		--run_name hicodet_single_run_000001 \
		--HOIDet \
		--validate \
		--share_enc \
		--pretrained_dec \
		--lr 1e-4 \
		--num_hoi_queries 16 \
		--set_cost_idx 20 \
		--hoi_act_loss_coef 10 \
		--hoi_eos_coef 0.1 \
		--temperature 0.2 \
    --epochs 30 \
		--no_aux_loss \
		--hoi_aux_loss \
		--dataset_file hico-det \
		--frozen_weights https://dl.fbaipublicfiles.com/detr/detr-r50-e632da11.pth \
		--data_path hico_20160224_det \
		--output_dir checkpoints/hico_det/

Not using distributed mode
Freeze weights for detector

[Logger] DETR Arguments:
	lr: 0.0001
	lr_backbone: 1e-05
	lr_drop: 80
	frozen_weights: https://dl.fbaipublicfiles.com/detr/detr-r50-e632da11.pth
	backbone: resnet50
	dilation: False
	position_embedding: sine
	enc_layers: 6
	dec_layers: 6
	num_queries: 100
	dataset_file: hico-det

[Logger] DETR_HOI Arguments:
	hoi_dec_layers: 6
	hoi_nheads: 8
	hoi_dim_feedforward: 2048
	hoi_idx_loss_coef: 1
	hoi_act_loss_coef: 10.0
	hoi_eos_coef: 0.1
	object_threshold: 0

[Logger] Number of params:  9902198

>>> Epoch #1
Epoch [ 1/30]  [ 3764/18816]  eta: 1:05:42  lr: 0.000100  loss: 99.4977 (136.4692)  loss_hidx: 3.1445 (3.2487)  loss_oidx: 3.7894 (3.7986)  loss_act: 6.8703 (11.1347)  loss_tgt: 2.5779 (4.3649)  loss_hidx_0: 3.0856 (3.2506)  loss_oidx_0: 3.7825 (3.8034)  loss_act_0: 7.0255 (11.4825)  loss_tgt_0: 2.5779 (4.4356)  loss_hidx_1: 3.1451 (3.2489)  loss_oidx_1: 3.8917 (3.8044)  loss_act_1: 6.6082 (11.4518)  loss_tgt_1: 2.5779 (4.4232)  lo